src/cli/subcommand/install.rs

```diff
@@ -105,16 +108,35 @@ impl CommandExecute for Install {
                 let builtin_planner = BuiltinPlanner::from_common_settings(settings)
                     .await
                     .map_err(|e| eyre::eyre!(e))?;
-                let res = builtin_planner.plan().await;
-                match res {
-                    Ok(plan) => plan,
-                    Err(e) => {
-                        if let Some(expected) = e.expected() {
-                            eprintln!("{}", expected.red());
-                            return Ok(ExitCode::FAILURE);
+                match existing_receipt {
+                    Some(existing_receipt) => {
+                        if existing_receipt.planner.typetag_name() != builtin_planner.typetag_name() {
+                            eprintln!("{}", format!("Found existing plan in `{RECEIPT_LOCATION}` which used a different planner, try uninstalling the existing install").red());
+                            return Ok(ExitCode::FAILURE)
                         }
-                        return Err(e.into())
-                    }
+                        if existing_receipt.planner.settings().map_err(|e| eyre!(e))? != builtin_planner.settings().map_err(|e| eyre!(e))? {
+                            eprintln!("{}", format!("Found existing plan in `{RECEIPT_LOCATION}` which used different planner settings, try uninstalling the existing install").red());
+                            return Ok(ExitCode::FAILURE)
+                        }
+                        if existing_receipt.actions.iter().all(|v| v.state == ActionState::Completed) {
+                            eprintln!("{}", format!("Found existing plan in `{RECEIPT_LOCATION}`, with the same settings, already completed, try uninstalling and reinstalling if Nix isn't working").red());
+                            return Ok(ExitCode::FAILURE)
+                        }
+                        existing_receipt
+                    } ,
```

Comment:
Kinda surprised the lint check doesn't care about this `} ,`

Reply:
Same :thinking: